horace_core/spaghetti_plot.m

```diff
@@ -109,426 +113,486 @@
-                labels{i} = sprintf('[%s]/[%s]',str_compress(num2str(hkl0(4:6)')),str_compress(num2str(hkls(1:3)')));
+                labels{i} = sprintf('[%s]/[%s]', str_compress(num2str(hkl0(4:6)')), str_compress(num2str(hkls(1:3)')));
             else
-                labels{i} = sprintf('%s/%s',labels{i},title(quotes(1)+1:quotes(2)-1));
+                labels{i} = sprintf('%s/%s', labels{i}, title(quotes(1)+1:quotes(2)-1));
             end
+        elseif isempty(quotes)
+            labels{i} = ['[', str_compress(num2str(hkls(1:3)'), ', '), ']'];
         else
-            if isempty(quotes)
-                labels{i} = ['[',str_compress(num2str(hkls(1:3)'),','),']'];
-            else
-                labels{i} = title(quotes(1)+1:quotes(2)-1);
-            end
+            labels{i} = title(quotes(1)+1:quotes(2)-1);
         end
+
         if isempty(quotes)
-            labels{i+1} = ['[',str_compress(num2str(hkls(4:6)'),','),']'];
+            labels{i+1} = ['[', str_compress(num2str(hkls(4:6)'), ', '), ']'];
         else
             labels{i+1} = title(quotes(3)+1:quotes(4)-1);
         end
     catch
         hkldir = wdisp_in(i).u_to_rlu(1:3, 1);
         inthkl = kron(wdisp_in(i).iint', wdisp_in(i).u_to_rlu(:, wdisp_in(i).iax));
-        hklcen = sum([mean(inthkl(1:3, [1 3]), 2) mean(inthkl(5:7, [2 4]), 2)], 2);
-        hkls = [wdisp_in(i).p{1}(1) * hkldir + hklcen; wdisp_in(i).p{1}(end) * hkldir + hklcen];
-        labels{i} = ['[',str_compress(num2str(hkls(1:3)'),','),']'];
-        labels{i+1} = ['[',str_compress(num2str(hkls(4:6)'),','),']'];
+        hklcen = sum([mean(inthkl(1:3, [1 3]), 2), mean(inthkl(5:7, [2 4]), 2)], 2);
+        hkls = [wdisp_in(i).p{1}(1) * hkldir + hklcen; ...
+                wdisp_in(i).p{1}(end) * hkldir + hklcen];
+        labels{i} = ['[', str_compress(num2str(hkls(1:3)'), ', '), ']'];
+        labels{i+1} = ['[', str_compress(num2str(hkls(4:6)'), ', '), ']'];
 
     end
+
     hkl0 = hkls;
 end
+
 % This might not be the best way to do this but the 2D dataset should(!) hopefully not
 %   take up too much memory to make a copy of wdisp...
 if opt.logscale
     for i=1:length(wdisp)
         wdisp(i).signal = log10(wdisp(i).signal);
     end
 end
+
 wdisp(1).title = title1(1:lnbrk);
+
 plot(wdisp);
 hold on;
-xrlp = zeros(length(wdisp)+1,1);
+
+xrlp = zeros(length(wdisp) + 1, 1);
+
 for i=1:length(wdisp)
     xrlp(i+1) = wdisp(i).x(end);
-    plot([1 1]*xrlp(i+1),get(gca,'YLim'),'-k');
+    plot([1 1] * xrlp(i+1), get(gca,'YLim'), '-k');
 end
+
 hold off;
+
 if ~isempty(opt.labels)
-    if iscellstr(opt.labels) && numel(opt.labels)==length(wdisp)+1
-        labels = opt.labels;
-    else
-        warning(['Not using user-supplied labels. They are either not a cell array of' ...
-            'strings or not enough for all segments']);
+    if ~iscellstr(opt.labels) || numel(opt.labels) ~= length(wdisp)+1
+        error('HORACE:spaghetti_plot:invalid_argument', ...
```

Comment:
Could we change this back to a `warning` rather than an `error`?

Reply:
I can, I put `error` in because that seems like somewhere you might not want to necessarily progress.